internal/shellgen/tmpl/glibc-patch.bash

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+set -euo pipefail
+
+declare -r pkg   # package that we're patching
+declare -r glibc # new glibc that we're patching in
+declare -r out   # nix output path that will contain the patched package
+
+# Paths to this script's dependencies set by nix.
+declare -r coreutils file findutils patchelf ripgrep
+
+# Explicitly declare the specific commands that this script depends on.
+hash -p "$coreutils/bin/cp" cp
+hash -p "$coreutils/bin/chmod" chmod
+hash -p "$coreutils/bin/cut" cut
+hash -p "$coreutils/bin/echo" echo
+hash -p "$coreutils/bin/head" head
+hash -p "$coreutils/bin/mktemp" mktemp
+hash -p "$coreutils/bin/rm" rm
+hash -p "$coreutils/bin/stat" stat
+hash -p "$coreutils/bin/wc" wc
+hash -p "$file/bin/file" file
+hash -p "$findutils/bin/find" find
+hash -p "$findutils/bin/xargs" xargs
+hash -p "$patchelf/bin/patchelf" patchelf
+hash -p "$ripgrep/bin/rg" rg
+
+# Copy the contents of the original package so we can patch them.
+cp -R "$pkg" "$out"
+
+# Find the new linker that we'll patch into all of the package's executables as
+# the interpreter.
+interp="$(find "$glibc/lib" -type f -maxdepth 1 -executable -name 'ld-linux-*.so*' | head -n1)"
+readonly interp
+
+patch() {
+	declare -r binary="$1" # ELF binary to patch
+	declare -r type="$2"   # "exe" or "lib"
+
+	declare set_interpreter
+	if [[ "$type" == "exe" ]]; then
+		set_interpreter=true
+	fi
+
+	perm=$(stat -c "%a" "$binary")
+	old_rpath="$(patchelf --print-rpath "$binary")"
+	new_rpath="$glibc/lib${old_rpath:+:$old_rpath}"
+
+	echo "running patchelf file=\"$binary\" type=\"$type\" rpath=\"$new_rpath\" perm=\"$perm\""
+	chmod u+w "$binary"
+	patchelf --set-rpath "$new_rpath" \
+	         --add-needed libBrokenLocale.so.1 \
```

Comment:
should `ld-linux-x86-64.so.2` (or equivalent for aarch64) also be in this list? Is there never a chance it is used as a library by some executable?

Reply:
I don't think it's necessary because the interpreter should just be using itself. If this turns out to be wrong I can add it back.